compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -310,6 +310,8 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         def compareThis = {
           val cls2 = tp2.cls
           tp1 match {
+            case tp1: ThisType =>
+              (tp1.cls eq cls2) && recur(tp1.tref, tp2.tref)
```

Comment:
I'm not sure about the condition here, `isSubPrefix` does something different:
https://github.com/lampepfl/dotty/blob/0525ba010061c060d759d869c63ea5b54415f729/compiler/src/dotty/tools/dotc/core/TypeComparer.scala#L859-L863
Should we do the same here and get rid of `isSubPrefix`, or is there a good reason for treating ThisType in prefix positions specially?

Reply:
The isSubPrefix logic is needed to deal with double vision with self types.

I think the test here can be simplified to
```scala
case tp1: ThisType => tp1.cls eq cls2
```
The `tref` is just used to carry the class.